server/ruby/config_helper.rb

```diff
@@ -0,0 +1,264 @@
+
+          Update `.env` with:
+
+          #{price_id}=#{stripe_price.id}
+        DOC
+      end
+      return false
+    else
+
+    end
+    true
+  end
+
+  def valid_domain?
+    domain = ENV['DOMAIN']
+    if domain.nil?
+      puts <<~DOC
+        No DOMAIN environment variable found in `.env`. This should be a
+        base URL and include the scheme like `http://localhost:4242`.
+      DOC
+      return false
+    end
+
+    if !domain.start_with?('http')
+      puts <<~DOC
+        Invalid DOMAIN set: `#{domain}`. This should be a
+        base URL and include the scheme like `http://localhost:4242`.
+      DOC
+      return false
+    end
+
+    true
+  end
+
+  def valid_paths?
+    static_dir = ENV['STATIC_DIR']
+    if static_dir.nil?
+      puts <<~DOC
+        No STATIC_DIR environment variable found in `.env`. This should be the
+        relative path to the directory where the client side HTML code is.
+      DOC
+      return false
+    end
+    static_dir_path = File.join(File.dirname(__FILE__), static_dir)
+    if !File.directory?(static_dir_path)
+      puts <<~DOC
+        The path to the STATIC_DIR is not a directory. Please check .env.
+      DOC
+      return false
+    end
+
+    if !File.exists?(File.join(static_dir_path, 'index.html'))
+      if static_dir == ''
+        puts <<~DOC
+          No value set for STATIC_DIR. If this sample was installed with the
+          Stripe CLI then STATIC_DIR is usually `../client`. If this sample was
+          git cloned, STATIC_DIR is typically set to `../../client`.
+        DOC
+        return false
+      else
+        puts <<~DOC
+          No `index.html` file found in #{static_dir_path}. Please check #{File.join(static_dir_path, 'index.html')}
+        DOC
+      end
+      return false
+    end
+
+    true
+  end
+
+  def valid_api_keys?
+    sk = ENV['STRIPE_SECRET_KEY']
+    if sk.nil? || (!sk.start_with?('sk_test_') && !sk.start_with?('rk_test_'))
+      puts <<~DOC
+        Your secret API key (STRIPE_SECRET_KEY) is configured incorrectly or
+        doesn't match the expected format. You can find your API keys in the Stripe
+        dashboard here: https://dashboard.stripe.com/test/apikeys. Then update
+        the .env file.
+      DOC
+      return false
+    end
+
+    pk = ENV['STRIPE_PUBLISHABLE_KEY']
+    if pk.nil? || !pk.start_with?('pk_test_')
+      puts <<~DOC
+        Your publishable API key (STRIPE_PUBLISHABLE_KEY) is configured incorrectly or
+        doesn't match the expected format. You can find your API keys in the Stripe
+        dashboard here: https://dashboard.stripe.com/test/apikeys. Then update
+        the .env file.
+      DOC
+      return false
+    end
+
+    pi = nil
+    begin
+      pi = Stripe::PaymentIntent.list({
+        limit: 1
+      }, {
+        api_key: sk
+      }).data.first
+    rescue => e
+      puts "Failed testing an API request with your STRIPE_SECRET_KEY `#{sk}` check `.env`: \n\n#{e}"
```

Comment:
I think it might be a good idea to truncate secrets in any messages in case the output is being logged somewhere unintended/unknown.